### PR TITLE
#114 add custom cost management

### PR DESCRIPTION
**Pull request by gluckzhang**
The custom cost management UI is available via `/infrawallet/custom_costs`. To enable custom costs support, the following configuration needs to be added:

```yaml
backend:
  infraWallet:
    integrations:
      custom:
        - name: <arbitrary_string>
```

Some screenshots:

image

image

image

### CODE REVIEW

File: plugins/infrawallet-backend/src/cost-clients/AwsClient.ts [hunk 1]
```diff
@@ -264,6 +264,7 @@ export class AwsClient extends InfraWalletClient {
                 service: this.convertServiceName(serviceName),
                 category: categoryMappingService.getCategoryByServiceName(this.provider, serviceName),
                 provider: this.provider,
+                provider_type: PROVIDER_TYPE.INTEGRATION,
```
Comment (emillg) [inline plugins/infrawallet-backend/src/cost-clients/AwsClient.ts:267]: providerType?

File: plugins/infrawallet-backend/src/models/CustomCost.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  const knex = await database.getClient();
+  const records = await knex('custom_costs').insert(data).returning('id');
+  return records.length;
+}
+
+// Get cost records by date range
+export async function getCustomCostsByDateRange(
+  database: DatabaseService,
+  start_date: Date,
+  end_date: Date,
```
Comment (emillg) [inline plugins/infrawallet-backend/src/models/CustomCost.ts:35]: endDate?

File: plugins/infrawallet/src/api/types.ts [hunk 3]
```diff
@@ -87,3 +87,21 @@ export type GetWalletResponse = {
...
 };
+
+export type CustomCost = {
+  id: string; // UUID
+  provider: string;
+  account: string;
+  service?: string;
+  category?: string;
+  currency: string;
+  amortization_mode: string; // average, as_it_is
```
Comment (emillg) [inline plugins/infrawallet/src/api/types.ts:98]: amortizationMode

Comment (emillg) [reply plugins/infrawallet/src/api/types.ts:98]: Is it because of database?

Comment (gluckzhang) [reply plugins/infrawallet/src/api/types.ts:98]: Yes it comes from the database table, I will transform the names :)

**Review by emillg — APPROVED** [review]

**Pull request merged**